proto/spacemesh/global_state_types.proto

```diff
@@ -0,0 +1,71 @@
+syntax = "proto3";
+package spacemesh;
+import "spacemesh/types.proto";
+
+// Data items related to an account
+enum AccountDataFlag {
+    NONE = 0;
+    TRANSACTION_RECIEPT = 1; // tx receipt for a tx to or from an account
+    REWARD = 2; // reward awarded to an account
+}
+
+message AccountDataParams {
+    AccountId account_id = 1;
+    uint32 account_data_bit_field = 2; // A bit field of AccountDataFlag
+}
+
+message AccountDataQueryParams {
+    AccountDataParams account_data = 1;
+    uint32 max_results = 2; // max numbers of results client would like to get
+    uint32 offset = 3; // return results from offset
+}
+
+// An account related data item
+message AccountData {
+    oneof item {
+        Reward reward = 1;
+        TransactionReceipt receipt = 2;
+        Account account = 3;
+    }
+}
+
+message AccountDataQueryResults {
+    repeated AccountData account_item = 1;
+    uint32 total_results = 2;
+}
+
+message SmesherDataQueryParams {
+    SmesherId smesher_id = 1;
+    uint32 max_results = 2; // max numbers of results client would like to get
+    uint32 offset = 3;      // return results from offset
+}
+
+message SmesherDataQueryResults {
+    repeated Reward rewards = 1;
+    uint32 total_results = 2;
+}
+
+enum GlobalStateEntityFlag {
+    NONE = 0;
+    REWARD = 1;
+    TRANSACTION_STATE = 2;
+    TRANSACTION_RECIEPT = 4;
+    GLOBAL_STATE = 8;
+}
+
+message GlobalStateEntities {
+    uint32 flags = 1; // bitmap of GlobalStateEntityFlags
+}
+
+message GlobalStateEntity {
+    oneof data {
+        Reward reward = 1;
+        TransactionState transaction_state = 2;
```

Comment:
Strictly speaking I don't think `TransactionState` belongs in the global state. Isn't it kinda orthogonal to global state? Cf. #36

Reply:
You are correct here. Looking at TransactionStateType I see it has one state that is STF related (PROCESSED) that we should remove and if we do that then it is only a mesh related tx state as it should be. I will fix it and remove it.

Reply:
But we have the receipt. If there is no receipt for a tx, then it was never executed by the STF. If there is, then we know its state.

I'm still not totally happy with TransactionState - I think it's very confusing the way it mixes state from a local node (REJECTED, CONFLICTING, etc.), with state from the mesh (PENDING), with post-STF global state (PROCESSED). Maybe let's discuss and work this out in #36?

Reply:
There is no way around it - because the journey to the mesh passes at these states. It correctly described these possible states. I've removed PROCESSED. So now, it just describes possible states from submission by user until it gets on the mesh. I don't think we want to refactor it into sub-states

Reply:
It should be in the mesh service and not in the global state service - I've moved it there.

Reply:
Also added `rpc TransactionsStateStream(TransactionsIds) returns (stream TransactionsState);` which is very useful for wallets to have.